roles/translator/src/lib/sv2/channel_manager/message_handler.rs

```diff
@@ -0,0 +1,426 @@
+                    range2,
+                )
+                .unwrap();
+                channel_manager_data.extranonce_prefix_factory =
+                    Some(Arc::new(Mutex::new(extended_extranonce_factory)));
+
+                let factory = channel_manager_data.extranonce_prefix_factory.as_ref().unwrap();
+                let new_extranonce_size = factory.safe_lock(|f| f.get_range2_len()).unwrap() as u16;
+                if downstream_extranonce_len <= new_extranonce_size as usize {
+                    let new_extranonce_prefix = factory
+                        .safe_lock(|f| f.next_prefix_extended(new_extranonce_size as usize))
+                        .unwrap()
+                        .unwrap()
+                        .into_b032();
+                    let new_downstream_extended_channel = ExtendedChannel::new(
+                        m.channel_id,
+                        user_identity.clone(),
+                        new_extranonce_prefix.clone().into_static().to_vec(),
+                        target.clone().into(),
+                        nominal_hashrate,
+                        true,
+                        new_extranonce_size,
+                    );
+                    channel_manager_data.extended_channels.insert(
+                        m.channel_id,
+                        Arc::new(RwLock::new(new_downstream_extended_channel)),
+                    );
+                    let new_open_extended_mining_channel_success = OpenExtendedMiningChannelSuccess {
+                        request_id: m.request_id,
+                        channel_id: m.channel_id,
+                        extranonce_prefix: new_extranonce_prefix,
+                        extranonce_size: new_extranonce_size,
+                        target: m.target.clone(),
+                    };
+                    return new_open_extended_mining_channel_success.into_static();
+                }
+            }
+
+            // If we are not in aggregated mode, we just insert the extended channel into the map
+            channel_manager_data.extended_channels
+                .insert(m.channel_id, Arc::new(RwLock::new(extended_channel)));
+
+            m.into_static()
+        }).unwrap();
+
+        self.channel_state
+            .sv1_server_sender
+            .send(Mining::OpenExtendedMiningChannelSuccess(success.clone()))
+            .await
+            .map_err(|e| {
+                error!("Failed to send OpenExtendedMiningChannelSuccess: {:?}", e);
+                HandlerError::External(Box::new(TproxyError::ChannelErrorSender))
+            })?;
+
+        Ok(())
+    }
+
+    async fn handle_open_mining_channel_error(
+        &mut self,
+        m: roles_logic_sv2::mining_sv2::OpenMiningChannelError<'_>,
+    ) -> Result<(), HandlerError> {
+        error!(
+            "Received OpenExtendedMiningChannelError from upstream with error code {}",
+            std::str::from_utf8(m.error_code.as_ref()).unwrap_or("unknown error code")
```

Comment:
here, we would only be showing error code, but not other relevant message fields

we have `Display` trait implementations for all messages now, we should leverage them
```suggestion
            "Received: {}", m
```

Reply:
and the `debug!` print below could be removed